server/memcached/src/main/java/org/infinispan/server/memcached/MemcachedBaseDecoder.java

```diff
@@ -106,12 +97,36 @@ protected final void exceptionCaught(Header header, Throwable t) {
       }
    }
 
+   private void initializeHandler() {
+      AdvancedCache<byte[], byte[]> c = createCache(server);
```

Comment:
This isn't really creating a cache right? Just retrieving it and thus shouldn't be blocking?

Reply:
Correct. At this point, the `CacheInitializerInboundAdapter` should have already executed. This method will also use the `getCache` in the server (`MemcachedServer` and `RespServer`, too), and it will throw an illegal state if the cache wasn't initialized before.